atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/KeyValueServices.java

```diff
@@ -196,4 +198,23 @@ public void run() {
         }
         return results;
     }
+
+    public static RowColumnRangeIterator mergeGetRowsColumnRangeIntoSingleIterator(KeyValueService kvs,
+                                                                                   TableReference tableRef,
+                                                                                   Iterable<byte[]> rows,
+                                                                                   ColumnRangeSelection columnRangeSelection,
+                                                                                   int batchHint,
+                                                                                   long timestamp) {
+        log.warn("This KVS does not support getRowsColumnRange with paging through all results simultaneously. Falling "
+                + "back to the less efficient per-row paging version.");
+        SizedColumnRangeSelection sizedColumnRangeSelection =
+                new SizedColumnRangeSelection(columnRangeSelection.getStartCol(),
+                                              columnRangeSelection.getEndCol(),
+                                              batchHint / Iterables.size(rows));
```

Comment:
We should probably make sure to bail if `rows.isEmpty()` to avoid divide by zero